cmds/pci/pci.go

```diff
@@ -23,8 +27,84 @@ var (
 	numbers    = flag.Bool("n", false, "Show numeric IDs")
 	dumpConfig = flag.Bool("c", false, "Dump config space")
 	devs       = flag.String("s", "*", "Devices to match")
+	format     = map[int]string{
+		32: "%08x:%08x",
+		16: "%08x:%04x",
+		8:  "%08x:%02x",
+	}
 )
 
+// maybe we need a better syntax than the standard pcitools?
+func registers(d pci.Devices, cmds ...string) {
+	var justCheck bool
+	for _, c := range cmds {
+		// TODO: replace this nonsense with a state machine.
+		// Split into register and value
+		rv := strings.Split(c, "=")
+		if len(rv) != 1 && len(rv) != 2 {
+			log.Printf("%v: only one = allowed. Due to this error no more commands will be issued", c)
+			justCheck = true
+			continue
+		}
+
+		// Split into register offset and size
+		rs := strings.Split(rv[0], ".")
+		if len(rs) != 1 && len(rs) != 2 {
+			log.Printf("%v: only one . allowed. Due to this error no more commands will be issued", rv[1])
+			justCheck = true
+			continue
+		}
+		s := 32
+		if len(rs) == 2 {
+			switch rs[1] {
+			default:
+				log.Printf("Bad size: %v. Due to this error no more commands will be issued", rs[1])
+				justCheck = true
+				continue
+			case "l":
+			case "w":
+				s = 16
+			case "b":
+				s = 8
+			}
+		}
+		if justCheck {
```

Comment:
I'm not sure I understand the justCheck... it's set to true below, but checked up here?

Reply:
you can go through this loop once per command. justCheck is set for any errors and subsequent commands are ignored.